Services/Administration/OJS.Services.Administration.Business/Problems/ProblemsBusinessService.cs

```diff
@@ -265,21 +270,39 @@ await this.transactionsProvider.ExecuteInTransaction(async () =>
         await this.submissionsCommonBusinessService.PublishSubmissionsForProcessing(submissions);
     }
 
-    private static void AddSubmissionTypes(Problem problem, ProblemAdministrationModel model)
+    private async Task AddSubmissionTypes(Problem problem, ProblemAdministrationModel model)
     {
-        var newSubmissionTypes = model.SubmissionTypes
-            .Select(x => new SubmissionTypeInProblem
+        var submissionTypeIds = model.SubmissionTypes.Select(st => st.Id).ToList();
+
+        var submissionTypes = await this.submissionTypesData
+            .All(st => submissionTypeIds.Contains(st.Id))
+            .ToDictionaryAsync(st => st.Id, st => st);
+
+        var newSubmissionTypes = new List<SubmissionTypeInProblem>();
+
+        foreach (var newSubmissionType in model.SubmissionTypes)
+        {
+            if (!submissionTypes.TryGetValue(newSubmissionType.Id, out var submissionType))
+            {
+                throw new BusinessServiceException($"Submission type with Id #{newSubmissionType.Id} does not exist.");
+            }
+
+            submissionType.IsSelectedByDefault = newSubmissionType.IsSelectedByDefault;
```

Comment:
If a problem has many submission types but must have 1 that always appears selected by default, wouldn't this make many submission types selected by default?

Reply:
No, the `frontend` implementation does not allow for more than one `default` submission type to be selected. `IsSelectedByDefault` will be true for only a single submission type for any given time.

Reply:
Is this handled on the backend as well? Legacy uses DefaultSubmissionTypeId in Problem.cs, this might be a better approach..

Reply:
I think that checking whether only one of all submission types has the property set to `true` is unnecessary.

Reply:
I guess we agreed on doing the implementation using `DefaultSubmissionTypeId` in `Problem`?

Reply:
Yes.